### PR TITLE
Bytes only rust

### PR DESCRIPTION
**Pull request by pawanjay176**
Making a PR so it'll be easy to squash merge.

### CODE REVIEW

File: bindings/rust/src/bindings.rs [hunk 1]
```diff
@@ -279,7 +225,11 @@ extern "C" {
     ) -> C_KZG_RET;
 }
 extern "C" {
-    pub fn blob_to_kzg_commitment(out: *mut KZGCommitment, blob: *mut u8, s: *const KZGSettings);
+    pub fn blob_to_kzg_commitment(
+        out: *mut KZGCommitment,
+        blob: *mut u8,
```
Comment (jtraglia) [inline bindings/rust/src/bindings.rs:230]: Can the type of this parameter be changed from `u8` to `Blob`? Would that compile?

File: bindings/rust/src/lib.rs [hunk 2]
```diff
@@ -222,20 +174,20 @@ impl KzgProof {
 
     pub fn verify_kzg_proof(
         &self,
-        kzg_commitment: KzgCommitment,
+        kzg_commitment: KZGCommitment,
         z: [u8; BYTES_PER_FIELD_ELEMENT],
         y: [u8; BYTES_PER_FIELD_ELEMENT],
```
Comment (jtraglia) [inline bindings/rust/src/lib.rs:179]: Could these be changed to `z: BLSFieldElement`?

File: bindings/rust/src/lib.rs [hunk 3]
```diff
@@ -257,65 +207,79 @@ impl KzgCommitment {
...
+    pub fn blob_to_kzg_commitment(mut blob: Blob, kzg_settings: &KZGSettings) -> Self {
+        let mut kzg_commitment: MaybeUninit<KZGCommitment> = MaybeUninit::uninit();
         unsafe {
-            bindings::blob_to_kzg_commitment(
-                kzg_commitment.as_mut_ptr(),
-                blob.as_mut_ptr(),
-                &kzg_settings.0,
-            );
-            Self(kzg_commitment.assume_init())
+            blob_to_kzg_commitment(kzg_commitment.as_mut_ptr(), blob.as_mut_ptr(), kzg_settings);
```
Comment (jtraglia) [inline bindings/rust/src/lib.rs:226]: Should the blob argument be `blob.as_ptr()`? The blob shouldn't be modified.

**Review by jtraglia — APPROVED** [review]
This looks great. Just a few little nits.

File: bindings/rust/src/lib.rs (hunk 3 above)
Comment (jtraglia) [reply bindings/rust/src/lib.rs:226]: Hmm, actually I'm not sure that would compile. I think you'd need to do something like `blobs.as_ptr() as *const u8`. There's are some other instances of `as_mut_ptr()` that maybe need to be changed too.

File: bindings/rust/src/bindings.rs (hunk 1 above)
Comment (jtraglia) [reply bindings/rust/src/bindings.rs:230]: Also, it doesn't need to be mutable, does it?

Comment (pawanjay176) [reply bindings/rust/src/bindings.rs]: Yeah i don't think it needs to. I think the c code at some point had it mutable, not sure.

**Comment by jtraglia** [comment]
LGTM 👍 thanks for making those changes.

**Pull request merged**